options.go

```diff
@@ -119,6 +120,26 @@ func WithBaseConfig(base *config.Config) ServerOption {
 	})
 }
 
+// WithDynamicConfigValue sets the given dynamic config key with the given set
+// of values. This will overwrite the key if already set.
+func WithDynamicConfigValue(key dynamicconfig.Key, value []dynamicconfig.ConstrainedValue) ServerOption {
+	return newApplyFuncContainer(func(cfg *liteconfig.Config) {
+		if cfg.DynamicConfig == nil {
+			cfg.DynamicConfig = dynamicconfig.StaticClient{}
+		}
+		cfg.DynamicConfig[key] = value
+	})
+}
+
+// WithSearchAttributeCacheDisabled disables search attribute caching. This
+// delegates to WithDynamicConfigValue.
+func WithSearchAttributeCacheDisabled() ServerOption {
+	return WithDynamicConfigValue(
+		dynamicconfig.ForceSearchAttributesCacheRefreshOnRead,
+		[]dynamicconfig.ConstrainedValue{{Value: true}},
+	)
+}
```

Comment:
Commented above, but should we make this the default and instead add an option for enabling the cache? We could comment to the effect of changing this default if/when we are able to remove the waiting period after registering search attributes, and if you need stable behavior use `WithDynamicConfigValue`.

Reply:
I think it should be opt-in, like persistence-disabled, especially now that we are going to turn it off by default on the CLI.